common/partition/default-partitioner.go

```diff
@@ -113,32 +110,10 @@ func (r *defaultPartitioner) pickIsolationGroup(wfPartition defaultWorkflowParti
 	if isAvailable {
 		return wfPartition.WorkflowStartIsolationGroup
 	}
-
-	// it's drained, fall back to picking a deterministic but random group
-	var availableList []string
```

Comment:
I think your change makes sense, and my thinking here wasn't very complete, it does remove the ability to explicitly drain a zone for a customer however. 

That might be ok, I am skeptical we're getting a lot value out of the zone drain feature, at least presently. Just double-checking that was your intent however?

Reply:
After reading through the rest of the PR, I think that your change is totally worthwhile and worth dropping the ability to drain isolation-groups at present. I think it implies the need for some more cleanup PRs later (which I'm happy to do) to cleanup the draining code, but availablity should be the first conconcern of matching and I think the first iteration failed at that, so this is preferable.

Reply:
We still can drain specific isolation groups because we also drain them at the poller side: https://github.com/cadence-workflow/cadence/blob/master/service/frontend/api/handler.go#L461 . 

You're correct that we're removing this functionality from the partitioner and from matching, and instead just relying on frontend to enforce it. So pollers that have already reached matching and are long-polling will be able to process a task but any subsequent polls from a drained group won't be able to pick up tasks.

Reply:
oh, that's totally fine then 🤦  forgot about that